Sources/ArcGISToolkit/Components/Offline/OfflineMapAreasView.swift

```diff
@@ -140,6 +169,37 @@ public struct OfflineMapAreasView: View {
             .frame(maxWidth: .infinity)
         }
     }
+    
+    private var offlineBannerView: some View {
+        Text("Network Offline")
+            .font(.footnote)
+            .foregroundStyle(.secondary)
+            .frame(maxWidth: .infinity)
+            .padding(EdgeInsets(top: 4, leading: 8, bottom: 4, trailing: 8))
+            .background(.ultraThinMaterial, ignoresSafeAreaEdges: [.bottom, .horizontal])
+    }
+    
+    private func view(for error: Error) -> some View {
+        VStack(alignment: .center) {
+            Image(systemName: "exclamationmark.circle")
+                .imageScale(.large)
+                .foregroundStyle(.red)
+            Text(error.localizedDescription)
+        }
+        .frame(maxWidth: .infinity)
+    }
+    
+    /// Loads the online and offline preplanned map models.
+    private func loadPreplannedMapModels() async {
+        await mapViewModel.loadPreplannedMapModels()
+        
+        switch mapViewModel.preplannedMapModels {
+        case .failure(_):
+            await mapViewModel.loadOfflinePreplannedMapModels()
+        default:
+            return
+        }
```

Comment:
Since only one case is being checked for and nothing is done in the other case, an if statement might be more appropriate.

```suggestion
        if case .failure = mapViewModel.preplannedMapModels {
            await mapViewModel.loadOfflinePreplannedMapModels()
        }
```